ui/src/app/shared/components/filter-drop-down.tsx

```diff
@@ -1,16 +1,49 @@
-import {DropDown} from 'argo-ui/src/components/dropdown/dropdown';
+import {Checkbox} from 'argo-ui';
+import * as classNames from 'classnames';
 import * as React from 'react';
-import {CheckboxList} from './checkbox-list';
+import {DropDown} from './dropdown/dropdown';
 
-export const FilterDropDown = (props: {values: {[label: string]: boolean}; onChange: (label: string, checked: boolean) => void}) => (
-    <DropDown
-        isMenu={true}
-        anchor={() => (
-            <div className={'top-bar__filter' + (props.values.size > props.values.size ? ' top-bar__filter--selected' : '')}>
-                <i className='argo-icon-filter' />
-                <i className='fa fa-angle-down' />
-            </div>
-        )}>
-        <CheckboxList values={props.values} onChange={(label, checked) => props.onChange(label, checked)} />
-    </DropDown>
-);
+interface FilterDropDownProps {
+    sections: FilterDropSection[];
+}
+
+export interface FilterDropSection {
+    title: string;
+    values: {[label: string]: boolean};
+    onChange: (label: string, checked: boolean) => void;
+}
+
+export const FilterDropDown = (props: FilterDropDownProps) => {
```

Comment:
This component is called `FilterDropDown`, but since it uses the abstract `DropDown` and adds filtering to it as content, I think it's a fair name.